src/index.js

```diff
@@ -406,3 +421,32 @@ async function prefetchUrls(documents) {
   }
   return result
 }
+
+function tryMatching(newDocs, oldDocs, L, LWP) {
+  let mismatch = 0
+  log('info', 'Trying to match olds and news')
+  for (const currentNewIndex in newDocs) {
+    const currentNew = newDocs[currentNewIndex]
+    let matches = []
+    for (const currentOld of oldDocs) {
+      const regex = new RegExp(`${currentNew.oldname}\\d{14}.pdf`)
+      const match = currentOld.filename.match(regex)
+      if (match) {
+        matches.push(match)
+      }
+    }
+    if (matches.length == 0 || matches.length > 1) {
+      mismatch++
```

Comment:
Matching is entangled with reporting here. I would separate the two.

```
const findMatch = (newDoc, oldDocs) => {

  return { newDoc: newDoc, matches }
}

const findMatches = (newDocs, oldDocs) => {
  const results = newDocs.map(newDoc => findMatch(newDoc, oldDocs))
  reportResults(results)
}

const reportResults = (matchResults) => { 
  const mismatches = results.filter(result => result.matches.length === 0)
  if (mismatches.length > 0) { // report something }
  const multiMatches = results.filter(result => result.matches.length > 1)
  if (multiMatches.length > 0) { // report something }
}
```

Reply:
Also it seems here that an oldDoc can be match several times. I think we should remove an oldDoc from the potential candidates for matching if we found a match for it.